src/modules/workspaces/workspaces.controller.ts

```diff
@@ -339,15 +347,106 @@ export class WorkspacesController {
   @ApiOkResponse({
     description: 'Created File',
   })
-  @UseGuards(WorkspaceGuard)
+  @UseGuards(WorkspaceGuard, SharingPermissionsGuard)
   @WorkspaceRequiredAccess(AccessContext.WORKSPACE, WorkspaceRole.MEMBER)
+  @RequiredSharingPermissions(SharingActionName.UploadFile)
   async createFile(
     @Param('workspaceId', ValidateUUIDPipe)
     workspaceId: WorkspaceAttributes['id'],
-    @UserDecorator() user: User,
+    @BehalfUserDecorator() behalfUser: User,
     @Body() createFileDto: CreateWorkspaceFileDto,
   ) {
-    return this.workspaceUseCases.createFile(user, workspaceId, createFileDto);
+    if (!behalfUser) {
+      throw new ForbiddenException('You have not permissions to do this!');
+    }
+
+    return this.workspaceUseCases.createFile(
+      behalfUser,
+      workspaceId,
+      createFileDto,
+    );
+  }
```

Comment:
We need to add logs in a future. When that time comes, we can bring back the @UserDecorator to get the actual requester and log from the controller (what I think is going to be the right way to do it).

Reply:
That's a great approach